activerecord/lib/active_record/persistence.rb

```diff
@@ -375,6 +375,7 @@ def upsert_all(attributes, on_duplicate: :update, update_only: nil, returning: n
       # how this "single-table" inheritance mapping is implemented.
       def instantiate(attributes, column_types = {}, &block)
         klass = discriminate_class_for_record(attributes)
+        attributes.except!(*klass.ignored_columns) if _has_attribute?(inheritance_column)
```

Comment:
Would it be a problem to ignore columns even if it's not an STI?

Reply:
No, but I believe it would be redundant because when creating an instance from a class that doesn't use Single Table Inheritance (STI), the attributes in 'ignored_columns' are correctly ignored. 
The issue arises only when the base class in an STI possesses all the attributes, but the child classes do not. This is because the base class requests all attributes in the database and instantiates the child classes with them, disregarding the 'ignored_columns' attribute

Reply:
- I think this should belong inside the [inheritance concern](https://github.com/rails/rails/blob/0b728682a8d2232751c652fed1b1f05ebdd81663/activerecord/lib/active_record/inheritance.rb) (could override `instantiate_instance_of` there with your change and call `super`).

- Should we use `using_single_table_inheritance?` instead of `_has_attribute?(inheritance...)`

Reply:
Nice suggestion! it makes so much sense 👌🏼

Reply:
The changes are ready for your review